doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,1060 @@
+  Default: 128,  range 0-240 
+- **config spanning_tree mst cost \<cost-value\>**
+  - Specify configuring the port level priority for root bridge in seconds.
+  Default: 0,  range 1-200000000 
+
+## Region Level
+Below commands allow configuring on region basis:
+
+- **config spanning_tree mst region name \<region-name\>**
+  - Edit the name of region
+  - region-name: Case sensitive, characters should be less than or equal to 32, default: mac-address of bridge
+- **config spanning-tree mst revision \<revision-number\>**
+  - Revision number is used to track changes in the configuration and to synchronize the configuration across the switches in the same region.
+  - revision-number: Default: 0, range: 0-65535
+
+## Instance Level
+
+Below commands allow configuration of an instance:
+
+- **config spanning_tree mst instance \<instance-id\> priority \<priority-value\>** 
+  - Configure priority of bridge for an instance.
+  - instance-id: id of the instance for which bridge priority is to be defined. If the provided instance id is not created yet, an error message is displayed.
+  - priority-value: Default: 32768, range: 0-61440 (should be multiple of 4096)
+- **config spanning_tree mst instance \<instance-id\> vlan (add|del)  \<vlan-id\>**
+  - VLAN to instance mapping.
+  - instance-id: id of the instance to which VLAN is to be mapped. If the provided instance id is not created yet, an error message is displayed.
+  - vlan-id: Range: 1-4094. If the provided VLAN is not created yet, an error message is displayed.
+  - Instance is only active when there is at least one VLAN member port configured for one of the mapped VLANs.
+
+## Instance, Interface Level
+Following commands are used for spanning-tree configurations on per instance, per interface basis:
+
+- **config spanning_tree mst instance \<instance-id\> interface \<ifname\> priority \<priority-value\>**
+  - Configure priority of an interface for an instance.
+  - priority-value: Default: 128, range: 0-240
+- **config spanning_tree mst instance \<instance-id\> interface <ifname\> cost \<cost-value\>**
+  - Configure path cost of an interface for an instance.
+  - cost-value: Range: 1-200000000
+
+## Interface Level
+Following new command will be added:
+
+- **config spanning_tree mst interface\<ifname\> {enable|disable}**
+  - Configure an interface for MSTP.
+- **config spanning_tree mst interface edgeport {enable|disable} \<ifname\>**
+  - Configure an interface as an edge port.
+  
+- **config spanning_tree interface \<ifname\> root_guard {enable|disable}**
+  - Configure an interface as a root_guard.
+
+- **config spanning_tree mst interface \<ifname\> bpdu_guard {enable|disable}**
+  - Configure an interface as a bpdu_guard. 
+
+- **config spanning_tree mst interface \<ifname\> guard {root|bpdu}**
+  - Configure an interface as a bpdu_guard. 
+
+Interface level CLI configurations of root guard, BPDU guard will also be supported for spanning-tree mode `mstp`.
+
```

Comment:
I see STP can be enabled on "portchannel" as well, so im assuming we may need to add CLI restrictions to ensure that when STP is enabled on a portchannel, the portchannel group cannot be deleted/modified. 

If thats true, please ensure its captured here or a different section of HLD

Reply:
Design will handle Portchannel modification/deletion automatically , so we don't plan to add any restrictions to prevent these operation on a STP enabled PortChannel.